### PR TITLE
bootctl: extend --make-machine-id-directory= documentation a bit

### PR DESCRIPTION
**Pull request by poettering**
The directory might not be created in the ESP but in the extended boot
loader partition, hence don#t claim otherwise.

Also, give a brief reason why the concept exists at all.

Link up machine-id man page.

### CODE REVIEW

File: man/bootctl.xml [hunk 1]
```diff
@@ -239,15 +239,25 @@
...
+        <option>remove</option>, respectively.  <literal>auto</literal> is equivalent to
+        <literal>yes</literal> if <filename>/etc/machine-id</filename> resides on a filesystem other than tmpfs
+        and <literal>no</literal> otherwise (in the latter case the machine ID is likely transient
+        only). Defaults to <literal>auto</literal>. See
+        <citerefentry><refentrytitle>machine-id</refentrytitle><manvolnum>5</manvolnum></citerefentry> for
+        details about the machine ID concept and file.</para>
+
+        <para>Overriding this may be desirable to hide the machine ID from the (unencrypted) ESP, configure a
+        <citerefentry><refentrytitle>kernel-install</refentrytitle><manvolnum>8</manvolnum></citerefentry>
+        script, or, conversely, persisently commit a transient machine ID.</para>
```
Comment (nabijaczleweli) [inline man/bootctl.xml]: Minor, but maybe just "or, conversely, commit a transient machine ID"? Seems fine otherwise.

File: man/bootctl.xml [hunk 2]
```diff
@@ -239,15 +239,25 @@
...
-        <para>Overriding this may be desired to hide the machine ID from the (unencrypted) ESP,
-        configure a <citerefentry><refentrytitle>kernel-install</refentrytitle><manvolnum>8</manvolnum></citerefentry> script, or,
-        conversely, commit a machine-id that only appears transient.</para></listitem>
+        <listitem><para>Control creation and deletion of the top-level machine ID directory on the file
+        system containing boot loader entries (i.e. beneath the file system returned by
+        <command>--print-boot-path</command> above) during <option>install</option> and
+        <option>remove</option>, respectively.  <literal>auto</literal> is equivalent to
+        <literal>yes</literal> if <filename>/etc/machine-id</filename> resides on a filesystem other than tmpfs
+        and <literal>no</literal> otherwise (in the latter case the machine ID is likely transient
+        only). Defaults to <literal>auto</literal>. See
```
Comment (nabijaczleweli) [inline man/bootctl.xml]: also, the "only" here reads a *touch* odd to me, but not remotely inexorably

**Comment by nabijaczleweli** [comment]
LGTM

**Comment by poettering** [comment]
thanks! setting green label.

**Pull request merged**